ext/gmp/gmp.c

```diff
@@ -1281,20 +1281,28 @@ ZEND_FUNCTION(gmp_fact)
 		RETURN_THROWS();
 	}
 
+#if SIZEOF_SIZE_T == 4
+	const zend_long maxbits = ULONG_MAX / GMP_NUMB_BITS;
+#else
+	const zend_long maxbits = INT_MAX;
+#endif
```

Comment:
This should be moved to the top-level, since we're already using it in `gmp_random_bits()`, and may want to re-use it elsewhere. (Could also be a macro; doesn't really matter, I think.)